### PR TITLE
feat(wallet): Implemented show all networks assets

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
- Implementing multi chain show balance (#27333)
- Refactor network selector screen to show new "All network" (#27074)
- Implemented multi-chain price and asset data collection (#27333)
- Show overlapping network icon on Portfolio and Asset selection screens (#27333)
- Show asset graph data for different networks and assets (#28748)
- Fixed unwanted duplicate removal from list (#28506)
- Fix balance issue for incoming asset
- Reduce latency by separating P3A calls
- Fixed edit visible asset to select assets
- Fixed edit visible assets list for all network to show default assets in "edit visible assets" when all networks is selected (until all network is supported)
- Updated portfolio, edit, view assets list UI list to show asset and network details from "All network" view
- Fix AssetDetails activity to use passed network
- Fixed AccountDetails activity issue with network
- Tested and added available icons for networks/native assets, logged issue for missing assets (#28798)
- Fix account list on asset details on AssetDetails screen
- Fixed cached NFTs after unlock wallet (#28301)
- Fixing asset graph data when multiple assets are shown
- Fix nested network list is instantly closed after selection (click) and improve selection (#27614)
- ERC721 NFTs with different token identifiers may show incorrect balance (#28627)
- Implement All networks list sort order
- Fix Buy, Send, Swap flows for selected multi-chain view token
- Fix crash with test network with new flow
- Ux: Show full network name on click on Edit visible via click popup
- Clear balance with network switching
- Clean up and refactoring to reuse

Resolves https://github.com/brave/brave-browser/issues/27074
Resolves https://github.com/brave/brave-browser/issues/27333
Resolves https://github.com/brave/brave-browser/issues/28506
Resolves https://github.com/brave/brave-browser/issues/28748
Resolves https://github.com/brave/brave-browser/issues/28627
Resolves https://github.com/brave/brave-browser/issues/28301
Resolves https://github.com/brave/brave-browser/issues/27614

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

[multi-chain.webm](https://user-images.githubusercontent.com/11755381/225301245-e26457d1-64ce-45f2-a163-7d0b69ec6387.webm)

### Additional:
- Not all network overlapping icons are show because they are unavailable: https://github.com/brave/brave-browser/issues/28561#issuecomment-1448109555
- Selected asset is preserved for long enough to work with incoming asset and single selection. Ideally will be showing no asset like desktop to avoid complication and extra calls to show the selected asset.

### CODE REVIEW

**Comment by SergeyZhukovsky** [comment]
Please don't combine if possible a huge amount of issues in one PR in the future. It would be hard to fix something in case of failure of one of the issues when verifying it, as it could be dependent or the worse case if we need to revert the whole PR(in that case everything has to be reverted). Plus to the above it's harder for reviewers to catch up with all of the issues and trying to understand what is going on.

**Comment by simoarpe** [comment]
😱 Reviewing 1200 LOC PR is every developer's worst nightmare.
Try to break the work into smaller PRs when possible. If not, split it into multiple commits with a meaningful name and a description to help the reviewers. 
I'm going to check out the branch and compile it locally.

**Comment by Pavneet-Sing** [comment]
> Please don't combine if possible a huge amount of issues in one PR in the future. It would be hard to fix something in case of failure of one of the issues when verifying it, as it could be dependent or the worse case if we need to revert the whole PR(in that case everything has to be reverted). Plus to the above it's harder for reviewers to catch up with all of the issues and trying to understand what is going on.

> scream Reviewing 1200 LOC PR is every developer's worst nightmare. Try to break the work into smaller PRs when possible. If not, split it into multiple commits with a meaningful name and a description to help the reviewers. I'm going to check out the branch and compile it locally.

Multi-chain is a feature and many things in the codebase had to be fixed to support this if you look thoroughly.
- https://github.com/brave/brave-browser/issues/27074, https://github.com/brave/brave-browser/issues/27333, https://github.com/brave/brave-browser/issues/28506, https://github.com/brave/brave-browser/issues/28748 all required for Multi-chain implementation
- https://github.com/brave/brave-browser/issues/28627 is also related with Portfolio and multi-chain, only has

https://github.com/brave/brave-core/blob/38ab4b20294c87ecb1400d2962ebb9ff3374ce5c/android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java#L316-L321

- https://github.com/brave/brave-browser/issues/27614 got automatically fixed with the MC implementation
- Multi-chain touches many different classes it is a significant PR and it's a big change/feature so everything has to go through.
- Additionally, we all have merged new features with big PRs because the codebase is not flexible or features often requires significant changes. MC touches everything so not sure why it is an issue now. Probably @SergeyZhukovsky can make a document and share in slack as a standard to be used a base going forward.


@simoarpe 
- The issues were found and fixed back and forth and it's easier to use squash to review and fix, but as mentioned above, other than the [NFT cache issue](https://github.com/brave/brave-browser/issues/28627), everything is related and saves a lot of time than maintaining a commit history.
- Probably, I can post and get help from other reviewers who are used to review large PR in brave and you can continue with other work if it gonna take considerable amount of time for you. Alternately, we split the files to review but don't think it is possible unless someone is used to it. LMK

Please feel free to discuss anything as I am using some new APIs with sophisticated flow so LMK if I can help with anything.

**Comment by SergeyZhukovsky** [comment]
@Pavneet-Sing so there is no way to split it and it has all be in a single PR and in a single commit right?

**Comment by bbondy** [comment]
Sorry for the drive by comment, but this comment came off as lacking in courtesy and I had to comment...

> Probably, I can post and get help from other reviewers who are used to review large PR in brave

It doesn't matter how great a reviewer is, if you have a very big commit that mixes several concerns, it will not be reviewed properly.  

It is not a matter of reviewer experience.

It is up to the person creating pull requests to split up a pull request's concerns into multiple digestible commits.  The author should create a story with their commits and have 1 concern per commit. 

What's more is that when combining multiple things it makes them insuperable. Maybe 1 concern of many needs special attention and needs an uplift. It could be cherry-picked if it was split up correctly, but cannot if it is buried in a big set of other changes.

Some sources, but it's not hard to find more:
- https://www.freecodecamp.org/news/optimize-pull-requests-for-reviewer-happiness/
- https://bitbucket.org/blog/the-pull-request-story
- https://www.swarmia.com/blog/why-small-pull-requests-are-better/?hsa_acc=6644081770&hsa_cam=19643106130&hsa_grp=145044313159&hsa_ad=646939742525&hsa_src=g&hsa_tgt=kwd-1940170916059&hsa_kw=best%20practice%20pull%20request&hsa_mt=p&hsa_net=adwords&hsa_ver=3
- https://levelup.gitconnected.com/how-to-write-the-perfect-pull-request-d044625ace98

**Comment by Pavneet-Sing** [comment]
> Sorry for the drive by comment, but this comment came off as lacking in courtesy and I had to comment...
> 
> > Probably, I can post and get help from other reviewers who are used to review large PR in brave
> 
> It doesn't matter how great a reviewer is, if you have a very big commit that mixes several concerns, it will not be reviewed properly.
> 
> It is not a matter of reviewer experience.
> 
> It is up to the person creating pull requests to split up a pull request's concerns into multiple digestible commits. The author should create a story with their commits and have 1 concern per commit.
> 
> What's more is that when combining multiple things it makes them inseparable. Maybe 1 concern of many needs special attention and needs an uplift. It could be cherry-picked if it was split up correctly, but cannot if it is buried in a big set of other changes.
> 
> Some sources, but it's not hard to find more:
> 
> * https://www.freecodecamp.org/news/optimize-pull-requests-for-reviewer-happiness/
> * https://bitbucket.org/blog/the-pull-request-story
> * https://www.swarmia.com/blog/why-small-pull-requests-are-better/?hsa_acc=6644081770&hsa_cam=19643106130&hsa_grp=145044313159&hsa_ad=646939742525&hsa_src=g&hsa_tgt=kwd-1940170916059&hsa_kw=best%20practice%20pull%20request&hsa_mt=p&hsa_net=adwords&hsa_ver=3
> * https://levelup.gitconnected.com/how-to-write-the-perfect-pull-request-d044625ace98



> > Probably, I can post and get help from other reviewers who are used to review large PR in brave
> 
> @Pavneet-Sing , there's no need. I'm always up for a challenge. Please don't take it the wrong way, small PRs should always be preferable, when I noticed the LOC were over a thousand it was natural for me expressing that as a candid observation.
> 
> At the end of the day, honest feedback is an invaluable tool that will make us better developers. I hope it didn't make you feel bad or anything, as your comment sounded a bit harsh. I just wanted to help.
> 
> I'm going to finish soon an initial review round 😀

I have been following the practices based on past PRs at brave to move forward quickly as often preferred. As I mentioned above, this was back and forth fixes so hard to maintain a history. Maintaining PR history is good this but I personally don't like to add random or partial commits. 

I will split this PR into 3-4 parts tomorrow. https://github.com/brave/brave-browser/issues/27074 will be in a `QA/No` PR as it won't be testable alone.

File: android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java [hunk 1]
```diff
@@ -212,22 +260,27 @@ public void refreshNetworks() {
         init();
     }
 
+    static void getAllNetworks(JsonRpcService jsonRpcService, List<Integer> supportedCoins,
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java:263]: package accessible only, if there is no modifier then it is `default`

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java [hunk 2]
```diff
@@ -314,104 +314,93 @@ private void setUpAccountList() {
...
+                                                Utils.setUpTransactionList(this, accountInfos,
+                                                        thisAssetItemModel, assetPrices,
+                                                        fullTokenList, nativeAssetsBalances,
+                                                        blockchainTokensBalances,
+                                                        findViewById(R.id.rv_transactions), this,
+                                                        mWalletTxCoinAdapter);
+
+                                                double thisPrice = Utils.getOrDefault(assetPrices,
+                                                        mAsset.symbol.toLowerCase(
+                                                                Locale.getDefault()),
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:343]: These parts are just auto formatted and didn't touched though have used and updated for the other parts.

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java [hunk 3]
```diff
@@ -314,104 +314,93 @@ private void setUpAccountList() {
...
+                                                        0.0d);
+                                                List<WalletListItemModel> walletListItemModelList =
+                                                        new ArrayList<>();
+                                                for (AccountInfo accountInfo : accountInfos) {
+                                                    final String accountAddressLower =
+                                                            accountInfo.address.toLowerCase(
+                                                                    Locale.getDefault());
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:350]: https://github.com/brave/brave-core/pull/17615#discussion_r1137491641

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java [hunk 4]
```diff
@@ -314,104 +314,93 @@ private void setUpAccountList() {
...
-                                                                        AssetDetailActivity.this));
-                                                    }
-                                                });
-                                    });
-                        });
+                                                    final String fiatBalanceString = String.format(
+                                                            Locale.getDefault(), "$%,.2f",
+                                                            thisPrice * thisAccountBalance);
+                                                    final String cryptoBalanceString =
+                                                            String.format(Locale.getDefault(),
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:369]: https://github.com/brave/brave-core/pull/17615#discussion_r1137491641

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/AndroidUtils.java [hunk 5]
```diff
@@ -98,4 +99,14 @@ private static void setViewVisibility(boolean isVisible, View... views) {
...
         return Resources.getSystem().getDisplayMetrics().heightPixels;
     }
+
+    /**
+     * Check if the fragment is safe to update its UI
+     * @param frag instance
+     * @return true if Fragment UI can be updated otherwise false
+     */
+    public static boolean canUpdateFragmentUi(Fragment frag) {
+        return !(frag.isRemoving() || frag.getActivity() == null || frag.isDetached()
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/AndroidUtils.java:109]: It's only for fragment itself not for the container or parent.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java [hunk 6]
```diff
@@ -70,4 +77,29 @@ public static boolean anyNull(Object... items) {
...
+        if (items == null) return "";
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < items.length; i++) {
+            String item = items[i];
+            if (TextUtils.isEmpty(item)) continue;
+            builder.append(item).append(separator);
+        }
+        // Delete separator after the last value
+        builder.deleteCharAt(builder.length() - 1);
+        return builder.toString();
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java:103]: I wanna delete the separator at last, and your code doesn't check for nulls or empty. 
BTW, the focus is not on refactoring things to streams. Probably just use it where it is more suitable but should not be enforced.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java [hunk 7]
```diff
@@ -102,23 +109,96 @@ public Double getMostRecentFiatSum() {
         return Double.valueOf(mFiatHistory[mFiatHistory.length - 1].price);
     }
 
+    public void fetchAllAssetsAndDetails(Callbacks.Callback1<PortfolioHelper> callback) {
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java:112]: cc @SergeyZhukovsky for thoughts, if we should split this and what's should be the ideal length of a method as this was mostly picked form the existing implementation.

**Comment by bbondy** [comment]
>  I personally don't like to add random or partial commits bit will do it for big PRs if that's what we prefer now.

I don't think I suggested to add random or partial commits. 
Please don't do that for large PRs. 

I did ask that you be courteous instead of defensive in review feedback, and that you split up concerns logically in a pull request in a way that makes sense to the reviewer and tells a story. Covering one major concern per commit when possible.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java [hunk 8]
```diff
@@ -102,23 +109,96 @@ public Double getMostRecentFiatSum() {
...
+        resetResultData();
+        if (mActivity.get() == null || mSelectedNetworks.isEmpty()) return;
+        int totalNetworks = mSelectedNetworks.size();
+        AtomicInteger resCount = new AtomicInteger();
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java:116]: because cannot use local variable as it is being used within the callback and don't wanna use a global variable to avoid logical complexity so this is a decent solution.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java [hunk 9]
```diff
@@ -102,23 +109,96 @@ public Double getMostRecentFiatSum() {
...
+                        if (resCount.incrementAndGet() == totalNetworks) {
+                            mUserAssetsList.sort(Comparator.comparing(
+                                    token -> mAssertSortPriorityPerCoinIndex.get(token.chainId)));
+                            for (AssetAccountsNetworkBalance assetAccountsNetworkBalance :
+                                    assetAccountsNetworkBalances) {
+                                // Sum across accounts
+                                for (AccountInfo accountInfo :
+                                        assetAccountsNetworkBalance.accountInfos) {
+                                    final String accountAddressLower =
+                                            accountInfo.address.toLowerCase(Locale.getDefault());
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java:143]: https://github.com/brave/brave-core/pull/17615#discussion_r1137492785

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java [hunk 10]
```diff
@@ -314,104 +314,93 @@ private void setUpAccountList() {
                 new WalletCoinAdapter(WalletCoinAdapter.AdapterType.ACCOUNTS_LIST);
         mWalletTxCoinAdapter =
                 new WalletCoinAdapter(WalletCoinAdapter.AdapterType.VISIBLE_ASSETS_LIST);
-        KeyringService keyringService = getKeyringService();
-        JsonRpcService jsonRpcService = getJsonRpcService();
-        if (keyringService != null && jsonRpcService != null && mWalletModel != null) {
-            keyringService.getKeyringInfo(
+        if (mWalletModel != null) {
+            mWalletModel.getKeyringModel().getKeyringPerId(
                     AssetUtils.getKeyringForCoinType(mCoinType), keyringInfo -> {
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:319]: https://github.com/brave/brave-core/pull/17615#discussion_r1137505578

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java:103]: I don't understand your answer.
I didn't add a `filter` to the chain as I just wanted to suggest a more elegant way, and give you the idea. Not sure about you mention the focus on refactoring the code using streams.

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:343]: > These parts are just auto formatted and didn't touched

I'm well aware, I hope you don't mind if I suggest minor improvements; feel free to skip them if they become too much.
Thank you for making the codebase better, I'm a huge supported of the Boy Scout rule: https://biratkirat.medium.com/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385

**Comment by Pavneet-Sing** [comment]
> > I personally don't like to add random or partial commits bit will do it for big PRs if that's what we prefer now.
> 
> I don't think I suggested to add random or partial commits. Please don't do that for large PRs.
> 
> I did ask that you be courteous instead of defensive in review feedback, and that you split up concerns logically in a pull request in a way that makes sense to the reviewer and tells a story. Covering one major concern per commit when possible.

As mentioned above, this solutions were implemented in iterations/fragments while implementing the other parts to test and move with the All network implementation.
Note: this commit summary of they draft is just a list of weekly updates on multi chain in prioritize manner.
Will follow this practice going forward to list and organise the issues/solutions.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java:103]: I have started using streams because now we can https://bravesoftware.slack.com/archives/C03MMJ4CHEZ/p1675178531068709 but this does not mean this we should use this everywhere possible to replace the existing working solution. Plus I also wanted to remove the char at last and the above solution is easy and more efficient than streams.

It's also good practice to not recommend partial or incomplete solutions without information. I think it is better to add subjective suggestions as nit and would recommend the same to separate important vs subjective feedbacks.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/AndroidUtils.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/AndroidUtils.java:109]: The function name itself explains what it does and I already added javadoc for you specially that should be enough. No need to add info about `isRemovingParent` as it is not related and needed. Would suggest to add minor or subjective review as nits because there are often many than the crucial ones.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java:103]: Thank you for the feedback and further elaboration @Pavneet-Sing. 
I completely agree with what you said about the use of Java streams.

It's important to remember that we're all human, and suggestions may not always be perfect, but that should not deter us from constructive criticism. It provides invaluable insights! 

Code reviews have many purposes, not only merging the code back to the main branch. They are an invaluable opportunity to discuss many things, just to name a few:
- Provide feedback and suggestions
- Creating connections with other team members
- Raising awareness of current coding styles and best practices

All of this becomes extremely hard (or impossible) if feedback is not accepted gracefully or (even worse) when the answers become rude. Demonstrating a courteous and receptive attitude towards feedback is crucial for a productive and harmonious workplace.
Really! I couldn't stress this point enough, we should try to be excellent to each other, and being courteous with those who review your code is the foundation for a fully functional team. 

Let's continue with this offline, feedback is an essential part of the development process, it plays a crucial part, but discussing it in PR comments may not be appropriate.
It's better to continue these conversations offline to prevent clogging up the PR comments section. Once again, thanks for your valuable input!

[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java:103]: I didn't deny the feedback or anything that you've mentioned above with a long paragraph and details, which often not needed. been seeing this patterns many times. The extra details was provided because it was asked.
I just clearly explained that we should not be tempted to use streams and should not enforce this in review, especially with partial solutions and full understanding. Would suggest to the same to take [things more gracefully](https://github.com/brave/brave-core/pull/17615#discussion_r1138442359).
